cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -278,6 +279,8 @@ private void writeProfile(ProfileId profileId, FullHttpRequest request)
     } catch (JsonSyntaxException e) {
       throw new BadRequestException("Unable to parse request body. Please make sure it is valid JSON", e);
     }
+    String totalWorkerCoreLabel = generateTotalWorkerCoreLabel(profileCreateRequest.getProvisioner());
+    profileCreateRequest.getProvisioner().setWorkerCoreLabel(totalWorkerCoreLabel);
```

Comment:
We shouldn’t expose worker labels to the SPI. Not all provisioner has the worker concept

Reply:
For such cases, we will use the default implementation of the label. 
Only the provisioners having some logic for worker cores  can implement that method .

Reply:
We probably need to use different naming. We are trying to describe total amount of processing power available. So, have it something like `totalProcessingCores` or `totalProcessingCPUs`. Worker is pretty specific, but some notion of cluster size in CPUs is applicable in general